Stripe/STPAPIClient.m

```diff
@@ -590,6 +594,34 @@ + (void)deleteSource:(NSString *)sourceID fromCustomerUsingKey:(STPEphemeralKey
 
 @end
 
+#pragma mark - ThreeDS2
+
+@implementation STPAPIClient (ThreeDS2)
+
+- (void)authenticate3DS2:(STDSAuthenticationRequestParameters *)authRequestParams
+        sourceIdentifier:(NSString *)sourceID
+              completion:(void (^)(STP3DS2AuthenticateResponse * _Nullable authenticateResponse, NSError * _Nullable error))completion {
+    NSString *endpoint = [NSString stringWithFormat:@"%@/authenticate", APIEndpoint3DS2];
+
+    NSMutableDictionary *appParams = [[STDSJSONEncoder dictionaryForObject:authRequestParams] mutableCopy];
+    appParams[@"deviceRenderOptions"] = @{@"sdkInterface": @"03",
+                                          @"sdkUiType": @[@"01", @"02", @"03", @"04", @"05"],
+                                          };
+    appParams[@"sdkMaxTimeout"] = @(5);
```

Comment:
should this get its value from `self->_currentAction.threeDSCustomizationSettings.authenticationTimeout` instead of hardcoded?

Reply:
oh yeah, thanks!